packages/core/src/Machine.ts

```diff
@@ -61,6 +62,18 @@ export function createMachine<
   },
   options?: Partial<MachineOptions<TContext, TEvent, TAction>>
 ): StateMachine<TContext, any, TEvent, TTypestate>;
+export function createMachine<
+  TContext,
+  TEvent extends EventObject,
+  TMeta extends TypegenMeta
+>(
+  // Ensure that only the first overload matches models, and prevent
+  // accidental inference of the model as the `TContext` (which leads to cryptic errors)
+  config: TContext extends Model<any, any, any, any>
+    ? never
+    : MachineConfig<TContext, any, TEvent> & { types: TMeta },
+  options?: Partial<MaybeTypegenMachineOptions<TContext, TEvent, TMeta>>
+): StateMachine<TContext, any, TEvent, any, any, TMeta>;
```

Comment:
We should try to have fewer overloads, not more - I wonder if there's a way we can add this to a different overload?

Reply:
We should deprecate (or straight up remove) `createMachine<typeof model>`. We can detect if a model is passed in through the original type (the one below this one) and warn.

Reply:
Let's straight up remove it. `createMachine<typeof model>()` is super easy to change to `model.createMachine()`.

Reply:
Added in #2657